### PR TITLE
Refer to jobs/forward model as forward model step in ert config

### PR DESCRIPTION
**Pull request by yngve-sk**
(internally only, does not alter any behavior)

**Issue**
Towards #7105

### CODE REVIEW

**Comment by xjules** [comment]
Some failures in tests
```sh
FAILED tests/unit_tests/config/test_ert_config.py::test_that_unknown_hooked_job_gives_config_validation_error - AssertionError: Regex pattern did not match.
 Regex: "Cannot setup hook for non-existing step name 'NO_SUCH_JOB'"
 Input: "/private/var/folders/3m/p59k4qdj0f17st0gn2cmj3640000gn/T/pytest-of-runner/pytest-0/popen-gw1/test_that_unknown_hooked_job_g0/test.ert: Line 3 (Column 15-26): Cannot setup hook for non-existing job name 'NO_SUCH_JOB'"
FAILED tests/unit_tests/config/test_forward_model.py::test_ert_config_throws_on_missing_forward_model_job - AssertionError: Regex pattern did not match.
```

File: src/ert/config/ert_config.py [hunk 1]
```diff
@@ -376,74 +380,74 @@ def _validate_dict(
...
         substitution_list: SubstitutionList,
         config_dict,
-    ) -> List[ForwardModel]:
+    ) -> List[ForwardModelStep]:
         errors = []
-        jobs = []
-        for job_description in config_dict.get(ConfigKeys.FORWARD_MODEL, []):
-            if len(job_description) > 1:
-                unsubstituted_job_name, args = job_description
+        steps = []
```
Comment (xjules) [inline src/ert/config/ert_config.py:390]: maybe to change `steps` -> `fm_steps`

File: src/ert/config/ert_config.py [hunk 2]
```diff
@@ -454,17 +458,17 @@ def forward_model_data_to_json(
         context = self.substitution_list
 
         class Substituter:
-            def __init__(self, job):
-                job_args = ",".join(
-                    [f"{key}={value}" for key, value in job.private_args.items()]
+            def __init__(self, step):
```
Comment (xjules) [inline src/ert/config/ert_config.py:461]: `fm_step`

File: src/ert/config/ert_config.py [hunk 3]
```diff
@@ -668,49 +672,49 @@ def _workflows_from_dict(
...
+                    f"Duplicate forward model step with name {name!r}, choosing "
+                    f"{step_config_file!r} over {steps[name].executable!r}",
                     name,
                 )
-            jobs[name] = new_job
+            steps[name] = new_step
 
-        for job_path in config_dict.get(ConfigKeys.INSTALL_JOB_DIRECTORY, []):
-            for file_name in _get_files_in_directory(job_path, errors):
+        for step_path in config_dict.get(ConfigKeys.INSTALL_JOB_DIRECTORY, []):
```
Comment (xjules) [inline src/ert/config/ert_config.py:697]: `fm_step_path`

File: src/ert/config/ert_config.py [hunk 4]
```diff
@@ -387,48 +387,48 @@ def _create_list_of_forward_model_steps_to_run(
         config_dict,
     ) -> List[ForwardModelStep]:
         errors = []
-        steps = []
+        fm_steps = []
         for step_description in config_dict.get(ConfigKeys.FORWARD_MODEL, []):
```
Comment (berland) [inline src/ert/config/ert_config.py:391]: Maybe do `step_*` -> `fm_step_*` also?

File: src/ert/cli/main.py [hunk 5]
```diff
@@ -49,8 +49,8 @@ def run_cli(args: Namespace, _: Any = None) -> None:
     # Create logger inside function to make sure all handlers have been added to
     # the root-logger.
     logger = logging.getLogger(__name__)
-    for job in ert_config.forward_model_list:
-        logger.info("Config contains forward model job %s", job.name)
+    for step in ert_config.forward_model_steps:
```
Comment (berland) [inline src/ert/cli/main.py:52]: step -> fm_step as in the next commit?

File: src/ert/cli/main.py [hunk 6]
```diff
@@ -49,8 +49,8 @@ def run_cli(args: Namespace, _: Any = None) -> None:
...
+        logger.info("Config contains forward model job %s", step.name)
```
Comment (berland) [inline src/ert/cli/main.py:53]: Not changing the logged string deliberate?

File: src/ert/config/forward_model.py [hunk 7]
```diff
@@ -9,7 +9,7 @@
 from .parse_arg_types_list import parse_arg_types_list
```
Comment (berland) [inline src/ert/config/forward_model.py:9]: Rename the file as well?

File: src/ert/simulator/forward_model_status.py [hunk 8]
```diff
@@ -119,7 +119,7 @@ def try_load(cls, path: str) -> "ForwardModelStatus":
         status = cls(status_data["run_id"], start_time, end_time=end_time)
 
         for job, state in zip(job_data["jobList"], status_data["jobs"]):
```
Comment (berland) [inline src/ert/simulator/forward_model_status.py:121]: job* -> fm_step (but maybe not the string "jobs" here as that might refer to jobs.json?)

File: tests/unit_tests/config/test_ert_config.py [hunk 9]
```diff
@@ -890,7 +890,9 @@ def test_that_unknown_job_gives_config_validation_error():
     with open(test_config_file_name, "w", encoding="utf-8") as fh:
         fh.write(test_config_contents)
 
-    with pytest.raises(ConfigValidationError, match="Could not find job 'NO_SUCH_JOB'"):
+    with pytest.raises(
+        ConfigValidationError, match="Could not find step 'NO_SUCH_JOB'"
```
Comment (berland) [inline tests/unit_tests/config/test_ert_config.py:894]: NO_SUCH_FORWARD_MODEL_STEP

File: src/ert/cli/main.py (hunk 6 above)
Comment (yngve-sk) [reply src/ert/cli/main.py:53]: Yeah, was not sure if it was a good idea wrt changing logs behavior

File: tests/unit_tests/config/test_ert_config.py [hunk 10]
```diff
@@ -1307,13 +1309,13 @@ def test_that_multiple_errors_are_shown_for_forward_model():
     expected_nice_messages_list = [
         (
             "test.ert: Line 3 (Column 15-29): "
-            "Could not find job 'does_not_exist' "
-            "in list of installed jobs: []"
+            "Could not find step 'does_not_exist' "
```
Comment (xjules) [inline tests/unit_tests/config/test_ert_config.py:1312]: when logging we should be explicit.

File: tests/unit_tests/config/test_forward_model.py [hunk 11]
```diff
@@ -24,7 +24,7 @@ def test_ert_config_throws_on_missing_forward_model_job(
             ["this-is-not-the-job-you-are-looking-for", "<WAVE-HAND>=casually"]
         )
 
-        with pytest.raises(expected_exception=ValueError, match="Could not find job"):
+        with pytest.raises(expected_exception=ValueError, match="Could not find step"):
```
Comment (xjules) [inline tests/unit_tests/config/test_forward_model.py:27]: `forward model step`

File: tests/unit_tests/config/test_forward_model_data_to_json.py [hunk 12]
```diff
@@ -356,34 +356,34 @@ def test_transfer_arg_types(context):
...
 @pytest.mark.usefixtures("use_tmpdir")
-def test_all_jobs(joblist, context):
-    run_all(joblist, context)
+def test_all_steps(steplist, context):
+    run_all(steplist, context)
 
 
 @pytest.mark.usefixtures("use_tmpdir")
-def test_various_null_fields(joblist, context):
+def test_various_null_fields(steplist, context):
```
Comment (xjules) [inline tests/unit_tests/config/test_forward_model_data_to_json.py:386]: `fm_step_list`

File: tests/unit_tests/config/test_forward_model_data_to_json.py [hunk 13]
```diff
@@ -237,11 +237,11 @@ def generate_job_from_dict(forward_model_config):
     return forward_model
 
 
-def set_up_forward_model(joblist) -> List[ForwardModel]:
-    return [generate_job_from_dict(job) for job in joblist]
+def set_up_forward_model(steplist) -> List[ForwardModelStep]:
```
Comment (xjules) [inline tests/unit_tests/config/test_forward_model_data_to_json.py:240]: `fm_step_list`

File: tests/unit_tests/config/test_forward_model_data_to_json.py [hunk 14]
```diff
@@ -22,7 +22,7 @@ def context():
 
 
 @pytest.fixture
-def joblist():
+def steplist():
```
Comment (xjules) [inline tests/unit_tests/config/test_forward_model_data_to_json.py:25]: `fm_step_list`

**Review by xjules — APPROVED** [review]
Nice job @yngve-sk ! :rocket: 
squash first then :)

**Pull request merged**